corehq/apps/userreports/tasks.py

```diff
@@ -204,37 +205,31 @@ def _run_report(backend_to_use):
     return objects
 
 
-@periodic_task(
-    run_every=crontab(minute="*/5"),
-    queue=settings.CELERY_PERIODIC_QUEUE,
-)
-def queue_async_indicators():
-    start = datetime.utcnow()
-    cutoff = start + ASYNC_INDICATOR_QUEUE_TIME
-    time_for_crit_section = ASYNC_INDICATOR_QUEUE_TIME.seconds - 10
+@periodic_task(run_every=crontab(minute="*/5"), queue=settings.CELERY_PERIODIC_QUEUE)
+def run_queue_async_indicators_task():
+    queue_async_indicators.delay()
 
+
+@serial_task('queue-async-indicators', timeout=10 * 60, queue=settings.CELERY_PERIODIC_QUEUE, max_retries=0)
+def queue_async_indicators():
     oldest_indicator = AsyncIndicator.objects.order_by('date_queued').first()
     if oldest_indicator and oldest_indicator.date_queued:
         lag = (datetime.utcnow() - oldest_indicator.date_queued).total_seconds()
         datadog_gauge('commcare.async_indicator.oldest_queued_indicator', lag)
 
-    with CriticalSection(['queue-async-indicators'], timeout=time_for_crit_section):
-        day_ago = datetime.utcnow() - timedelta(days=1)
-        indicators = AsyncIndicator.objects.all()[:settings.ASYNC_INDICATORS_TO_QUEUE]
-        if indicators:
-            lag = (datetime.utcnow() - indicators[0].date_created).total_seconds()
-            datadog_gauge('commcare.async_indicator.oldest_created_indicator', lag)
-        indicators_by_domain_doc_type = defaultdict(list)
-        for indicator in indicators:
-            # don't requeue anything htat's be queued in the past day
-            if not indicator.date_queued or indicator.date_queued < day_ago:
-                indicators_by_domain_doc_type[(indicator.domain, indicator.doc_type)].append(indicator)
-
-        for k, indicators in indicators_by_domain_doc_type.items():
-            now = datetime.utcnow()
-            if now > cutoff:
-                break
```

Comment:
do we still need something like this to make sure this doesn't run over the 10 minute timeout, or will this task always finish in under 10 minutes?